### PR TITLE
Docs - contribution additions

### PR DESCRIPTION
**Pull request by hepiyellow**

### CODE REVIEW

File: docs/contribution/COMPONENT_STRUCTURE.md [hunk 1]
```diff
@@ -21,33 +22,49 @@ wix-style-react
...
+### index.js
 1. Every component should have an index file  exporting the default entry for easy importing.
 
-## ComponentName.js
+### ComponentName.js
 1. The actual component. A component can be either a simple standalone component or composition of other components.
 1. Read the [Component Guidelines section](./COMPONENT_GUIDELINES.md) for more information.
 
-## ComponentName.scss
+### ComponentName.scss
```
Comment (argshook) [inline docs/contribution/COMPONENT_STRUCTURE.md]: line 12 says `.st.css` but here `.scss`, i guess here should also say `.st.css`?

File: docs/contribution/STYLING.md [hunk 2]
```diff
@@ -5,9 +5,12 @@
 
 ## Best practices
 1. For RTL support, add rules under `:global .rtl { }`.
+1. The convension for allowing styling of the component's root element is by exposing a `className` property. (This applies out-of-the-box when using Stylable).
```
Comment (argshook) [inline docs/contribution/STYLING.md]: typo `convention`

File: docs/contribution/TESTING.md [hunk 3]
```diff
@@ -20,30 +20,43 @@ describe('Checkbox', () => {
...
-1. Tests pages are the actual documentation done in `storybook`.
+
+#### File Structure
+
 1. Every component will have the test file next to it with the convention of `ComponentName.e2e.js`.
 1. Every component uses and expose a **driver**, to help interacting with the component. Read more about drivers [here](./TEST_DRIVERS.md). The driver naming convention is `ComponentName.protractor.driver.js`
 
+#### Test pages
+1. Tests pages are the actual documentation done in `storybook`.
+1. You may run you test on the story's Playground, or on it's examples.
```
Comment (argshook) [inline docs/contribution/TESTING.md]: typo `your`

File: docs/contribution/TESTING.md [hunk 4]
```diff
@@ -20,30 +20,43 @@ describe('Checkbox', () => {
...
+1. Sometimes you don't want to have tidious testing examples in the documentation story, so you may create a Test story by the name `ComponentTestStory.js` and add to to the `Tests` category in the storybook. Use:
```
Comment (argshook) [inline docs/contribution/TESTING.md]: typo `tedious` :)

File: docs/contribution/TESTING.md [hunk 5]
```diff
@@ -52,7 +65,77 @@ eyes.it('should test something with a screenshot on demand', async () => {
...
+
+  const driver = buttonTestkitFactory({dataHook: 'storybook-button'});
+
+  beforeAll(async () => {
+    await browser.get(storyUrl);
+    await waitForVisibilityOf(driver.element(), 'Cannot find Button');
+  });
+
+  afterEach(async () => {
+    await autoExampleDriver.reset();
```
Comment (argshook) [inline docs/contribution/TESTING.md]: i think we should suggest only `remount` and don't even show `reset`

File: docs/contribution/STYLING.md (hunk 2 above)
Comment (shlomitc) [inline docs/contribution/STYLING.md]: remove mentioning `stylable`

File: docs/contribution/STYLING.md [hunk 6]
```diff
@@ -5,9 +5,12 @@
...
 
 ## Next Generation
 1. In some components, we use [`Stylable`](https://stylable.io/). Stylable enables you to expose a style API that maps its internal parts so you can reuse components across teams without sacrificing stylability.
 We use it in very few component now, but this in general will be the next generation of styling of the library.
 
+When creating a new component, please use Stylable.
```
Comment (shlomitc) [inline docs/contribution/STYLING.md]: please consider using `stylable`

File: docs/contribution/TESTING.md [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
 ### Component (Unit) Tests
 1. Tests are running with [`jest`](https://facebook.github.io/jest/) in and `JSDOM` environment.
-1. Test in this API level are ones the require browser-like environment but can still run without any visual rendering. The nature of these tests is testing the behavior of a component and wiring methods. For example: clicking on a component triggers a callback, changing the input value, etc...
+1. Test in this API level are ones that require browser-like environment but can still run without any visual rendering. The nature of these tests is testing the behavior of a component and wiring methods. For example: clicking on a component triggers a callback, changing the input value, etc...
```
Comment (shlomitc) [inline docs/contribution/TESTING.md]: Test**s**

File: docs/contribution/TESTING.md [hunk 8]
```diff
@@ -20,30 +20,43 @@ describe('Checkbox', () => {
...
+
 1. Every component will have the test file next to it with the convention of `ComponentName.e2e.js`.
 1. Every component uses and expose a **driver**, to help interacting with the component. Read more about drivers [here](./TEST_DRIVERS.md). The driver naming convention is `ComponentName.protractor.driver.js`
 
+#### Test pages
+1. Tests pages are the actual documentation done in `storybook`.
+1. You may run you test on the story's Playground, or on it's examples.
+1. Sometimes you don't want to have tidious testing examples in the documentation story, so you may create a Test story by the name `ComponentTestStory.js` and add to to the `Tests` category in the storybook. Use:
+```js
+import {getTestStoryKind} from '../storyHierarchy`
```
Comment (shlomitc) [inline docs/contribution/TESTING.md]: should explain what is `storyHierarchy`,
also you miss the `'` after the `y` char.

File: docs/contribution/TESTING.md [hunk 9]
```diff
@@ -20,30 +20,43 @@ describe('Checkbox', () => {
...
 1. Tests are running with [`protractor`](http://www.protractortest.org/#/) which runs in actual `chrome` browser.
 1. Visual regression tests are done with [`eyes`](https://github.com/wix/eyes.it) (powered by applitools).
-1. Tests pages are the actual documentation done in `storybook`.
+
+#### File Structure
+
 1. Every component will have the test file next to it with the convention of `ComponentName.e2e.js`.
 1. Every component uses and expose a **driver**, to help interacting with the component. Read more about drivers [here](./TEST_DRIVERS.md). The driver naming convention is `ComponentName.protractor.driver.js`
 
+#### Test pages
```
Comment (shlomitc) [inline docs/contribution/TESTING.md]: put it below the visual testing section

File: docs/contribution/TESTING.md [hunk 10]
```diff
@@ -52,7 +65,77 @@ eyes.it('should test something with a screenshot on demand', async () => {
   // do other manipulations
 });
 ```
+#### Writing E2E Tests
```
Comment (shlomitc) [inline docs/contribution/TESTING.md]: please move this section with all the example to a separate markdown. it becomes too specific and too long

File: docs/contribution/COMPONENT_GUIDELINES.md [hunk 11]
```diff
@@ -1,9 +1,35 @@
...
+## Structure
+
+Every component folder should be a single component.
+
+## DataHook
+
 1. Every component should accept a `dataHook` property in order to allow it to be tested. A component can extend `WixComponent` in order to be a pure component and get `dataHook` for free.
-1. Exposing `className` property is allowed.
+
+1. If you use Stylable, then using the the Stylable's StyleSheet function (usually named `style()`), will pass a `data-hook` prop onto the dom.
```
[the remark this answers is not in the document]
Comment (hepiyellow) [reply docs/contribution/COMPONENT_GUIDELINES.md]: I think it is an important information about the "magic" that goes on with how we use Stylable.
We can have a separate "Stylable-Guidelines" page , so it doesn't clutter this one.

File: docs/contribution/COMPONENT_STRUCTURE.md [hunk 12]
```diff
@@ -1,18 +1,19 @@
 # Component structure
 The following structure is the basic structure for a component.
 
+## Component Specific Files
```
[the remark this answers is not in the document]
Comment (hepiyellow) [reply docs/contribution/COMPONENT_STRUCTURE.md]: There is another section later which is :
"## Common Files"

as opposed to :
"## Specific Files"

Any idea for naming this better?

**Comment by hepiyellow** [comment]
@shlomitc welcome to review changes.

**Pull request merged**